mods/action.js

```diff
@@ -6,47 +6,174 @@
+    'articleexists': errDefs['400'],
+    'badformat': errDefs['400'],
+    'badmd5': errDefs['400'],
+    'badtoken': errDefs['400'],
+    'invalidparammix': errDefs['400'],
+    'invalidsection': errDefs['400'],
+    'invalidtitle': errDefs['400'],
+    'invaliduser': errDefs['400'],
+    'missingparam': errDefs['400'],
+    'missingtitle': errDefs['400'],
+    'nosuchpageid': errDefs['400'],
+    'nosuchrcid': errDefs['400'],
+    'nosuchrevid': errDefs['400'],
+    'nosuchsection': errDefs['400'],
+    'nosuchuser': errDefs['400'],
+    'notext': errDefs['400'],
+    'notitle': errDefs['400'],
+    'pagecannotexist': errDefs['400'],
+    'revwrongpage': errDefs['400'],
+    /* 401 - unauthorised */
+    'cantcreate-anon': errDefs['401'],
+    'confirmemail': errDefs['401'],
+    'noedit-anon': errDefs['401'],
+    'noimageredirect-anon': errDefs['401'],
+    'protectedpage': errDefs['401'],
+    'readapidenied': errDefs['401'],
+    /* 403 - access denied */
+    'autoblocked': errDefs['403'],
+    'blocked': errDefs['403'],
+    'cantcreate': errDefs['403'],
+    'customcssjsprotected': errDefs['403'],
+    'customcssprotected': errDefs['403'],
+    'customjsprotected': errDefs['403'],
+    'emptynewsection': errDefs['403'],
+    'emptypage': errDefs['403'],
+    'filtered': errDefs['403'],
+    'hookaborted': errDefs['403'],
+    'noedit': errDefs['403'],
+    'noimageredirect': errDefs['403'],
+    'permissiondenied': errDefs['403'],
+    'protectednamespace': errDefs['403'],
+    'protectednamespace-interface': errDefs['403'],
+    'protectedtitle': errDefs['403'],
+    'readonly': errDefs['403'],
+    'unsupportednamespace': errDefs['403'],
+    'writeapidenied': errDefs['403'],
+    /* 409 - conflict */
+    'cascadeprotected': errDefs['409'],
+    'editconflict': errDefs['409'],
+    'pagedeleted': errDefs['409'],
+    'spamdetected': errDefs['409'],
+    /* 413 - body too large */
+    'contenttoobig': errDefs['413'],
+    /* 429 - rate limit exceeded */
+    'ratelimited': errDefs['429'],
+    /* 501 - not supported */
+    'editnotsupported': errDefs['501']
+};
+function apiError(apiErr) {
+    var ret;
+    apiErr = apiErr || {};
+    ret = {
+        message: 'MW API call error ' + apiErr.code,
+        status: errDefs['500'].status,
+        body: {
+            type: errDefs['500'].type,
+            title: apiErr.code || 'MW API Error',
+            description: apiErr.info || 'Unknown MW API error'
+        }
+    };
+    if(apiErr.code && errCodes.hasOwnProperty(apiErr.code)) {
+        ret.status = errCodes[apiErr.code].status;
+        ret.body.type = errCodes[apiErr.code].type;
+    }
+    return new rbUtil.HTTPError(ret);
+}
+
+/**
+ * Action module code
+ */
 function ActionService (options) {
     this.apiURITemplate = options.apiURI;
 }
 
 ActionService.prototype.apiURI = function(domain) {
+    // Re-map test domain
+    if(domain === 'en.wikipedia.test.local') { domain = 'en.wikipedia.org'; }
```

Comment:
Actually, we might not need this rewriting any more. The test config just ignores the domain for the API URI.